os/drivers/lcd/mipi_lcd.c

```diff
@@ -38,6 +38,75 @@
 #define CONFIG_LCD_MAXPOWER 100
 #endif
 
+#if defined(CONFIG_LCD_ST7701SN) && (defined(CONFIG_LCD_LANDSCAPE) || defined(CONFIG_LCD_RLANDSCAPE))	//Two lcd buffers to avoid screen tearing
+#define NUM_OF_LCD_BUFFER	2
+uint8_t *lcd_buffer[NUM_OF_LCD_BUFFER] = { NULL, NULL };
+int lcd_buffer_index = 0;
+
+#if defined(CONFIG_LCD_LANDSCAPE)
+static void lcd_rotate_buffer(short int* src, short int* dst)
+{
+	int row;
+	int col;
+	int dst_inc = 2 * CONFIG_LCD_XRES;
+	short int val0;
+	short int val1;
+	short int val2;
+	short int val3;
+	short int *psrc;
+	short int *pdst;
+
+	for (row = 0; row < CONFIG_LCD_XRES; row += 2) {
+		psrc = src + row * CONFIG_LCD_YRES;
+		pdst = dst + CONFIG_LCD_XRES - row - 2;
+		for (col = 0; col < CONFIG_LCD_YRES; col += 2) {
+			val0 = *(psrc + 0);
+			val1 = *(psrc + 1);
+			val2 = *(psrc + CONFIG_LCD_YRES + 0);
+			val3 = *(psrc + CONFIG_LCD_YRES + 1);
+			psrc += 2;
+			*(pdst + 0) = val2;
+			*(pdst + 1) = val0;
+			*(pdst + CONFIG_LCD_XRES) = val3;
+			*(pdst + CONFIG_LCD_XRES + 1) = val1;
+			pdst += dst_inc;
+		}
+	}
+}
+
+#elif defined(CONFIG_LCD_RLANDSCAPE)
+static void lcd_rotate_buffer(short int* src, short int* dst)
+{
+	int row;
+	int col;
+	int dst_inc = 2 * CONFIG_LCD_XRES;
+	short int val0;
+	short int val1;
+	short int val2;
+	short int val3;
+	short int *psrc;
+	short int *pdst;
+
+	for (row = 0; row < CONFIG_LCD_XRES; row += 2) {
+		psrc = src + row * CONFIG_LCD_YRES;
+		pdst = dst + row + (CONFIG_LCD_YRES - 1) * CONFIG_LCD_XRES;
```

Comment:
How about adding #ifdef here to reduce duplicate code?

Reply:
I have added the #ifdef already.